### PR TITLE
Factor out SYMPY_INTERP

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92234
* #94201
* __->__ #94307

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/_dynamo/guards.py [hunk 1]
```diff
@@ -637,13 +635,8 @@ def direct_negation(a, b):
                 ("___check_tensors", check_tensors_fn),
                 ("___check_tensors_verbose", check_tensors_verbose_fn),
                 ("tensor_check_names", tensor_check_names),
-                ("floor", math.floor),
-                ("ceiling", math.ceil),
-                ("Eq", direct_equality),
-                ("Ne", direct_negation),
-                ("Mod", sympy.Mod),
```
Comment (albanD) [inline torch/_dynamo/guards.py:644]: Many of these new point to different functions (no use of direct_equality, sympy.Mod, etc) is that expected?

Comment (ezyang) [reply torch/_dynamo/guards.py:644]: yes. The use of sympy.Mod is bad

**Review by albanD — APPROVED** [review]

File: torch/_dynamo/guards.py [hunk 2]
```diff
@@ -637,13 +635,8 @@ def direct_negation(a, b):
                 ("___check_tensors", check_tensors_fn),
                 ("___check_tensors_verbose", check_tensors_verbose_fn),
                 ("tensor_check_names", tensor_check_names),
-                ("floor", math.floor),
-                ("ceiling", math.ceil),
-                ("Eq", direct_equality),
```
Comment (voznesenskym) [inline torch/_dynamo/guards.py:642]: Delete direct equality and negation? These were just to remap and it looks like we aren't using them anymore?

**Comment by voznesenskym** [comment]
Oh PR is closed nvm

**Pull request closed without merging**